### ISSUE TITLE
Doesn't emit errors on compile errors, just outputs to console

### ISSUE THREAD

**Issue by frankandrobot** [ivogabe/gulp-typescript#359]
The use case is that we're actually wrapping the gulp task in a Promise.
So the code looks something like this:

```javascript
return new Promise((resolve, reject) => {

    const stream = gulp.src([//...])
      .pipe(gulpTs({
        //...
      }))
      .pipe(gulp.dest(//...))

    stream.on('end', resolve)
    stream.on('error', reject)
})
```
```sh
# console logs but no error event
master\jspm\etp-ios-bridge\src\BofaiOSBridge.ts(143,23): error TS2339: Property 'assign' does not exist on type 'ObjectConstructor'.
[02:30:44] TypeScript: 1 semantic errors
[02:30:44] TypeScript: emit failed
```

**Unfortunately, the error event never fires when there's a compile error.** All we see are console errors. 
As a result, the Promise never errors out. 

The use case is that this Promise is part of a deployment task. So obviously, we don't want to deploy broken builds.

**Comment by ivogabe** [comment]
You should add the listener directly after gulp-typescript. These errors aren't piped sadly.
```javascript
return new Promise((resolve, reject) => {

    const stream = gulp.src([//...])
      .pipe(gulpTs({
        //...
      }))
      .on('error', reject)
      .pipe(gulp.dest(//...))

    stream.on('end', resolve)
    stream.on('error', reject)
})
```

**Comment by frankandrobot** [comment]
thanks
